ethereumj-core/src/main/java/org/ethereum/manager/WorldManager.java

```diff
@@ -121,9 +124,25 @@ private void init() {
         syncManager.init(channelManager, pool);
     }
 
+
+    @Override
+    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        this.ctx = applicationContext;
+    }
+
+    /**
+     * @param listener
+     * @deprecated use #subscription instead.
+     */
+    @Deprecated
     public void addListener(EthereumListener listener) {
         logger.info("Ethereum listener added");
-        ((CompositeEthereumListener) this.listener).addListener(listener);
+        ((BackwardCompatibilityEthereumListenerProxy) listener).getCompositeListener().addListener(listener);
```

Comment:
If user have used our listener in most common way (`ethereum.addListener(new EthereumListenerAdapter(){})`) he will get strange exception here, I guess.

Reply:
Why do you think so ? This code fully support old logic.

Reply:
Users have code like
`ethereum.addListener(new EthereumListenerAdapter());`
after that it goes:
-> 
`worldManager.addListener(listener)`
->
`((BackwardCompatibilityEthereumListenerProxy) EthereumListenerAdapter instance)`